### ISSUE TITLE
Should we blend declaration-block-* category and the rule-set-* categories?

### ISSUE THREAD

**Issue by davidtheclark** [stylelint/stylelint#91]
They seem close enough to me that we could just blend them and reduce some complexity.

Is there a reason I'm missing?

**Comment by MoOx** [comment]
I guess it depends on the type of rules we are creating. Something like `declaration-block-trailing-semicolon` do not make more sense cause it cannot be applied to rule-set. Right ?

**Comment by jeddy3** [comment]
Also, I think declaration block is more specific than rule-set I.e. rule sets can nest whereas declaration blocks can't. For example, given:

```css
@print {
  a { color: pink }
  b { color: red }
}
```

I think `rule-set-no-duplicate-propeties` will flag a warning, whereas `declaration-block-no-duplicates` won't.

Or I might of got the distinction wrong?

Excuse typos, on phone.

**Comment by davidtheclark** [comment]
The only difference between a "declaration block" and a "rule set" is that the rule set contains a selector. So rule sets could also be considered to have optional trailing semicolons --- the same last semicolon as the declaration block (I don't think @ai made a mistake in the PostCSS API to have [`rule.semicolon`](https://github.com/postcss/postcss/blob/master/docs/api.md#rulesemicolon) instead of creating another `DeclarationBlock` class). And @jeddy3 in your example you actually have two rule sets, not one: I'm pretty sure that every declaration block is part of a rule set.

**Comment by jeddy3** [comment]
> And @jeddy3 in your example you actually have two rule sets, not one: I'm pretty sure that every declaration block is part of a rule set.

I think I got a bit muddled with this one. I'm happy to merge the two together, perhaps under the shorter label of `rule` as to match the postcss api. e.g.

* `declaration-block-no-duplicate-properties` -> `rule-no-duplicate-properties`
* `declaration-block-properties-order` -> `rule-properties-order`
* `rule-set-no-single-line` -> `rule-no-single-line`

A little clearer for the end-user too?

**Comment by davidtheclark** [comment]
I think it should be done as you describe, @jeddy3.

**Comment by MoOx** [comment]
:+1:

**Comment by jeddy3** [comment]
Groovy. I'll rename things later today.
